src/robot/libraries/BuiltIn.py

```diff
@@ -613,9 +613,17 @@ def should_be_equal(self, first, second, msg=None, values=True):
         for example, string ``false`` or ``no values``. See `Boolean arguments`
         section for more details.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``first`` and ``second`` should be compared case-insensitively,
+        provided that ``first`` and ``second`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         If both arguments are multiline strings, the comparison is done using
         `multiline string comparisons`.
         """
+        if is_truthy(ignore_case) and is_string(first) and is_string(second):
+            first = first.lower()
+            second = second.lower()
```

Comment:
Adding workarounds like that is annoying, but the framework working incorrectly on some platform is worse. It's especially bad if tests fail on that platform when run on CI. In this particular case we can argue that people ought to use newish IronPython versions where this problem is fixed. Like we don on our CI.